lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -369,82 +369,242 @@ void LCIStrategicPlugin::handleGreenSignalScenario(const cav_srvs::PlanManeuvers
 }
 
 
-TrajectoryParams LCIStrategicPlugin::handleFailureCaseHelper(double starting_speed, double departure_speed,  double speed_limit, double remaining_downtrack, double remaining_time, double traffic_light_downtrack)
+TrajectoryParams LCIStrategicPlugin::handleFailureCaseHelper(const lanelet::CarmaTrafficSignalPtr& traffic_light, double current_time, double starting_speed, double departure_speed,  double speed_limit, double remaining_downtrack, double remaining_time, double traffic_light_downtrack)
 {
   //Requested maneuver needs to be modified to meet remaining_dist req
   //by trying to get close to the target_speed and remaining_time as much as possible
-  TrajectoryParams params;
-
-  params.is_algorithm_successful = false;
-  params.case_num = CASE_1;
+  TrajectoryParams return_params;
+  TrajectoryParams traj_upper;
+  TrajectoryParams traj_lower;
 
   ROS_DEBUG_STREAM("HANDLE_LAST_RESORT_CASE: Starting...");
-  double modified_remaining_time;
+  double starting_downtrack = traffic_light_downtrack - remaining_downtrack;
+  double modified_remaining_time_upper; // upper meaning downtrack vs time trajectory is curved upwards
+  double modified_remaining_time_lower; // upper meaning downtrack vs time trajectory is curved lower
+  double modified_departure_speed_upper;
+  double modified_departure_speed_lower;
+  bool calculation_success_upper = true; // identifies places in codes where calculation can be invalid such as negative distance
+
+  // TODO change light_controlled_intersection_tactical_plugin...
+
+  // the upper ET
+  // accel case
+  traj_upper.t0_ = current_time;
+  traj_upper.v0_ = starting_speed;
+  traj_upper.x0_ = starting_downtrack;
+  traj_upper.is_algorithm_successful = false;
+  traj_upper.case_num = CASE_1;
+
+  if (departure_speed > starting_speed)
+  {
+    if ((pow(departure_speed,2) - pow(starting_speed,2))/(2*max_comfort_accel_) >= remaining_downtrack)
+    {
+      ROS_DEBUG_STREAM("Failure case upper 1");
```

Comment:
It is not technically a failure case. It just means that the vehicle cannot reach to departure speed which is ok. You can say, HandleFailureCase -> Upper Trajectory -> Actual speed less than desired departure speed.